regparser/layer/terms.py

```diff
@@ -187,101 +97,20 @@ def is_exclusion(self, term, node):
-                            (pos_start, pos_start + len(included_term)))
-
-        if stack and self.has_parent_definitions_indicator(stack):
-            for match, _, _ in grammar.smart_quotes.scanString(node.text):
-                term = match.term.tokens[0].lower().strip(',.;')
-                #   Don't use pos_end because we are stripping some chars
-                pos_start = match.term.pos[0]
-                add_match(node,
-                          term,
-                          (pos_start, pos_start + len(term)))
-
-        for match, _, _ in grammar.scope_term_type_parser.scanString(
-                node.text):
-            # Check that both scope and term look valid
-            if (self.scope_of_text(match.scope, Label.from_node(node),
-                                   verify_prefix=False)
-                    and re.match("^[a-z ]+$", match.term.tokens[0])):
-                term = match.term.tokens[0].strip()
-                pos_start = node.text.index(term, match.term.pos[0])
-                add_match(node, term, (pos_start, pos_start + len(term)))
-
-        if hasattr(node, 'tagged_text'):
-            for match, _, _ in grammar.xml_term_parser.scanString(
-                    node.tagged_text):
-                """Position in match reflects XML tags, so its dropped in
-                preference of new values based on node.text."""
-                for match in chain([match.head], match.tail):
-                    pos_start = self.pos_start_excluding(
-                        match.term.tokens[0], node.text,
-                        included_defs + excluded_defs)
-                    term = node.tagged_text[
-                        match.term.pos[0]:match.term.pos[1]].lower()
-                    match_len = len(term)
-                    add_match(node,
-                              term,
-                              (pos_start, pos_start + match_len))
-
-        return included_defs, excluded_defs
-
-    def pos_start_excluding(self, needle, haystack, exclusions):
-        """Search for the first instance of `needle` in the `haystack`
-        excluding any overlaps from `exclusions`. Implicitly returns None if
-        it can't be found"""
-        start = 0
-        while start >= 0:
-            start = haystack.find(needle, start)
-            if not any(r.position[0] <= start and r.position[1] >= start
-                       for r in exclusions):
-                return start
-            start += 1
-
-    def subpart_scope(self, label_parts):
-        """Given a label, determine which sections fall under the same
-        subpart"""
-        reg = label_parts[0]
-        section = label_parts[1]
-        for subpart in self.subpart_map:
-            if section in self.subpart_map[subpart]:
-                return [[reg, sect] for sect in self.subpart_map[subpart]]
-        return []
+        references = []
+        for finder in (def_finders.ExplicitIncludes(),
```

Comment:
Interesting idea! I don't know that that can work here as the instances of finders have construction parameters, but I think we can use the idea to our benefit elsewhere.